src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,132 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_userAgeLess18YearsOld_notOk() {
+        User user = new User("user", "123456", 17);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_userAgeNegative_notOk() {
+        User user = new User("user", "123456", -1);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_passwordLessThanSixSymbol_notOk() {
+        User user = new User("user", "12345", 18);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_suchLoginAlreadyExistedInTheStorage_notOk() {
+        User loginAlreadyExist = new User("LoginAlreadyExist", "123456", 19);
+        registrationService.register(loginAlreadyExist);
```

Comment:
lets add user directly to storage (in case register method does not work correctly)